tests/cell/cell_test_suite.cpp

```diff
@@ -849,6 +853,328 @@ class cell_test_suite : public test_suite
+
+        {
+            xlnt::workbook wb_full;
+            auto fmt_full = wb_full.create_format();
+
+            // Border
+            xlnt::border test_border;
+            test_border.side(xlnt::border_side::top, xlnt::border::border_property().style(xlnt::border_style::thick).color(xlnt::color::blue()));
+            fmt_full.border(test_border, true);
+
+            // Protection
+            fmt_full.protection(xlnt::protection().locked(false).hidden(true), true);
+
+            // Custom number format (non-builtin)
+            xlnt::number_format custom_nf("0.000%");
+            fmt_full.number_format(custom_nf, true);
+
+            // Pivot button and quote prefix
+            fmt_full.pivot_button(true);
+            fmt_full.quote_prefix(true);
+
+            // Style
+            auto src_style = wb_full.create_style("TestStyle");
+            src_style.font(xlnt::font().name("Arial"), true);
+            fmt_full.style("TestStyle");
+
+            cell_dest_full.format(fmt_full);
+        }
+        // Source workbook destroyed, verify format is still valid
+
+        xlnt_assert(cell_dest_full.has_format());
+        xlnt_assert(cell_dest_full.format().border_applied());
+        xlnt_assert(cell_dest_full.border().side(xlnt::border_side::top).is_set());
+        xlnt_assert_equals(cell_dest_full.border().side(xlnt::border_side::top).get().style().get(), xlnt::border_style::thick);
+        xlnt_assert(cell_dest_full.format().protection_applied());
+        xlnt_assert_equals(cell_dest_full.protection().locked(), false);
+        xlnt_assert_equals(cell_dest_full.protection().hidden(), true);
+        xlnt_assert(cell_dest_full.format().number_format_applied());
+        xlnt_assert_equals(cell_dest_full.number_format().format_string(), "0.000%");
+        xlnt_assert(cell_dest_full.format().pivot_button());
+        xlnt_assert(cell_dest_full.format().quote_prefix());
+        xlnt_assert(cell_dest_full.has_style());
+        xlnt_assert_equals(cell_dest_full.style().name(), "TestStyle");
+
+        // Test 6: cell::value() does shallow copy within same workbook
+        xlnt::workbook wb_same;
+        auto ws_same = wb_same.active_sheet();
+        auto cell_source = ws_same.cell("A1");
+        auto cell_dest = ws_same.cell("B1");
+
+        cell_source.value("Test");
+        cell_source.font(xlnt::font().bold(true).size(14));
+
+        auto initial_format_count = wb_same.format_count();
+
+        cell_dest.value(cell_source); // Copy within same workbook
+
+        // Format count should NOT increase (shallow copy)
+        xlnt_assert_equals(wb_same.format_count(), initial_format_count);
+
+        // Both cells should share the same format object
+        xlnt_assert(cell_dest.has_format());
+        xlnt_assert_equals(cell_dest.font().bold(), true);
+        xlnt_assert_equals(cell_dest.font().size(), 14.0);
+
+        // Test 7: cell::value() clones format from different workbook (automatic deep-copy)
```

Comment:
Maybe we can also check whether the format count is indeed increased in the target workbook. This would cleanly indicate that the format is indeed deep copied into the new workbook.